services/src/main/java/org/keycloak/services/resources/admin/AdminAuth.java

```diff
@@ -59,14 +59,9 @@ public AccessToken getToken() {
 
 
     public boolean hasRealmRole(String role) {
-        if (client instanceof ClientModel) {
-            RoleModel roleModel = realm.getRole(role);
-            if (roleModel == null) return false;
-            return user.hasRole(roleModel) && client.hasScope(roleModel);
-        } else {
-            AccessToken.Access access = token.getRealmAccess();
-            return access != null && access.isUserInRole(role);
-        }
+        RoleModel roleModel = realm.getRole(role);
```

Comment:
This change misses the branch when `client` is `null` and thus makes breaking changes.

Reply:
I looked in all code path and I thought client can never be null.